tests/python/pants_test/base_test.py

```diff
@@ -302,6 +303,15 @@ def context(self, for_task_types=None, for_subsystems=None, options=None,
       if scope is None:
         raise TaskError('You must set a scope on your task type before using it in tests.')
       optionables.add(task_type)
+      # If task is expected to inherit goal-level options, register those directly
+      # on the task, by subclassing the goal options registrar and overriding its scope.
+      if issubclass(task_type, GoalOptionsMixin):
```

Comment:
Rather than inspecting the mixin, what about just giving all Tasks a default `goal_options_registrar_cls = None`?

Reply:
If we want to get away from tying tasks to their goals, then we need to move in the opposite direction, I think... I'll figure something out.